addons/sale_management/models/sale_order.py

```diff
@@ -62,6 +62,13 @@ def _compute_option_data_for_template_change(self, option):
             'discount': option.discount,
         }
 
+    def update_pricelist(self):
+        self.ensure_one()
+        res = super().update_pricelist()
+        for line in self.sale_order_option_ids:
+            line.price_unit = self.pricelist_id.with_context(uom=line.uom_id.id).get_product_price(line.product_id, line.quantity, self.partner_id)
```

Comment:
Prefer to give the uom_id as method argument than through the context ;).
`            line.price_unit = self.pricelist_id.get_product_price(line.product_id, line.quantity, self.partner_id, uom_id=line.uom_id.id)
`

Reply:
@Feyensv  Done.